### ISSUE TITLE
In-order access to LinkedHashMap keys

### ISSUE THREAD

**Issue by dmolesUC3** [javaslang/javaslang#1850]
Presumably for reasons of efficiency, `LinkedHashMap.keySet()` returns the unordered `keySet` of the underlying `HashMap`.  It would be convenient to be able to get the keys in order without having to resort to a workaround like `map.iterator().map(entry -> entry._1)`, even if the underlying implementation isn't any less ugly than that. E.g.

```
/** Returns the keys contained in this map, in insertion order. */
public Queue<K> keys() {
  return list.map(entry -> entry._1);
}
```

(Forgive me if there's a more idiomatic way to do this; I have some Scala background but I've been stuck in Ruby for a year or two, and my Java and Javaslang are both rusty.)

**Comment by ruslansennov** [comment]
We can't change `LinkedHashMap::keys` signature as you proposed, because it will clash with method in super interface. However we could change it to something like this
```java
    @Override
    public Set<K> keySet() {
        return new LinkedHashSet(this);
    }
```
This would keep order of elements

**Comment by dmolesUC3** [comment]
That's why I suggested `keys()` rather than `keySet()`, but the `LinkedHashSet` idea is much, much better — I forgot that a `LinkedHashSet` is backed by a `LinkedHashMap`.

To do that, though, wouldn't you also need to change the `LinkedHashSet` constructor to take a `LinkedHashMap<T, Object>` rather than a `LinkedHashMap<T, T>`?

**Comment by ruslansennov** [comment]
> That's why I suggested adding `keys()` rather than changing `keySet()`

Oh, sorry :)

>  To do that, though, wouldn't you also need to change the `LinkedHashSet` constructor to take a `LinkedHashMap<T, Object>` rather than a `LinkedHashMap<T, T>`?

Yes, wrapped map should be changed to 
```java
private final LinkedHashMap<T, Object> map;
```

**Comment by ruslansennov** [comment]
`java.util.LinkedHashMap.keySet()` also keeps order

**Comment by danieldietrich** [comment]
Fixed with #1854
